app/policies/poll_policy.rb

```diff
@@ -2,6 +2,16 @@
 
 class PollPolicy < ApplicationPolicy
   def vote?
-    StatusPolicy.new(current_account, record.status).show? && !current_account.blocking?(record.account) && !record.account.blocking?(current_account)
+    viewable_through_normal_policy? && accounts_not_blocking?
+  end
+
+  private
+
+  def viewable_through_normal_policy?
+    StatusPolicy.new(current_account, record.status).show?
+  end
+
+  def accounts_not_blocking?
+    !current_account.blocking?(record.account) && !record.account.blocking?(current_account)
```

Comment:
Unrelated to this PR - I wonder how often this "account blocking or blocked by other account" check is done this way, and if often enough, does that merit a method on Account or Block.